### PR TITLE
Refactor search

### PR DESCRIPTION
**Pull request by alecgibson**
This change completely removes the `Search` class. This class perviously had knowledge of several areas of the domain (content, audits, themes). The main aim of the change is to replace this with smaller classes whose knowledge is more isolated.

The change is made in a number of atomic commits, which:

* Introduce the replacements for `Search`: `Content::Query` and `Audits::ContentQuery`
* Replace all instances of `Search` with the new classes
* Remove the `Search` class entirely, and move the unit tests to cover the new classes
* Remove the `Results` class, whose usefulness has all but disappeared with the removal of `Search`

### CODE REVIEW

File: spec/domain/audits/content_query_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+RSpec.describe Audits::ContentQuery do
+  let!(:content_item) { create(:content_item) }
+  let!(:audit) { create(:audit, content_item: content_item) }
+
+  it "can filter by audit status" do
+    subject.audit_status(:audited)
+    expect(subject.scope).to eq [content_item]
```
Comment (andrewgarner) [inline spec/domain/audits/content_query_spec.rb]: The `contain_exactly` matcher would work well here – https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher

```rb
expect(subject.scope).to contain_exactly(content_item)
```

File: spec/domain/content/query_spec.rb [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    it "can paginate" do
+      subject
+        .per_page(5)
+        .page(2)
+
+      results = subject.scope
+
+      expect(results.total_pages).to eq(2)
+      expect(results.count).to eq(1)
```
Comment (andrewgarner) [inline spec/domain/content/query_spec.rb]: How about this:

```rb
expect(subject.scope).to have_attributes(total_pages: 2, count: 1)
```

File: spec/domain/content/query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    it "can filter by title" do
+      foo = create(:content_item, title: "barfoobaz")
+      subject.title("foo")
+      expect(subject.scope).to eq [foo]
+    end
+  end
+
+  describe "with 26 content items" do
+    let!(:content_items) do
+      (1..26).map { create(:content_item) }
```
Comment (andrewgarner) [inline spec/domain/content/query_spec.rb]: You can do this:

```rb
create_list(: content_item, 26)
```

File: spec/domain/audits/content_query_spec.rb (hunk 1 above)
Comment (alecgibson) [inline spec/domain/audits/content_query_spec.rb]: Yeah, fine. I prefer these, but I largely just copy/pasted these tests.

**Comment by alecgibson** [comment]
@andrewgarner I've addressed your comments.

**Comment by alecgibson** [comment]
[Refactor Search](https://trello.com/c/eQ4lKUE3/387-refactor-search)

**Comment by pmanrubia** [comment]
Hi Alec, Thank you for your work on splitting content Search and Auditing. 

This first step moves our code base to a better position. There are a few things that I would like to explore with you in incoming PRs: having individual queries for auditing and doing an implicit use of scope.

In incoming PRs we will also be moving the queries out of the controllers, creating operations/commands to isolate the user needs, and fulfilling them via the Domain layer.

Good to merge

**Pull request merged**